components/overflow-group/overflow-group.js

```diff
@@ -0,0 +1,495 @@
+
+		this._mini = this.openerType === OPENER_TYPE.ICON || swapToMiniButton;
+		this._chompIndex = this._overflowMenuHidden ? null : showing.count;
+
+		this.dispatchEvent(new CustomEvent('d2l-overflow-group-updated', { bubbles: true, composed: true }));
+	}
+	_convertToDropdownItem(node) {
+		const tagName = node.tagName.toLowerCase();
+		let menuItem;
+		if (tagName === 'd2l-button' || tagName === 'd2l-button-subtle' || tagName === 'button' || tagName === 'd2l-button-icon') {
+			menuItem = this._createMenuItem(node);
+		} else if (tagName === 'd2l-link' || tagName === 'a') {
+			menuItem = this._createMenuItemLink(node);
+		} else if (node.getAttribute('role') === 'separator') {
+			menuItem = this._createMenuItemSeparator();
+		} else if (tagName === 'd2l-menu' || tagName === 'd2l-dropdown' || tagName === 'd2l-dropdown-button' || tagName === 'd2l-dropdown-button-subtle') {
+			menuItem = this._createMenuItemMenu(node);
+		} else if (tagName === 'd2l-menu-item') {
+			// if the menu item has children treat it as a menu item menu
+			if (node.children.length > 0) {
+				menuItem = this._createMenuItemMenu(node);
+			} else {
+				menuItem = this._createMenuItem(node);
+			}
+		} else if (node.classList.contains('d2l-overflow-group-custom-item')) {
+			menuItem = this._createMenuItem(node);
+		} else {
+			return;
+		}
+		return menuItem;
+	}
+	_createMenuItem(node) {
+		const childText = node.text || node.firstChild && (node.firstChild.label || node.firstChild.text || node.firstChild.textContent.trim());
+		const disabled = node.disabled;
+		return html`<d2l-menu-item 
+			?disabled=${disabled}
+			text="${childText}">
+		</d2l-menu-item>`;
+	}
+	_createMenuItemLink(node) {
+		const preventDefault = node.dataPreventDefault;
+		const text =  node.textContent.trim();
+		const href =  node.href;
+		const target = node.target;
+
+		return html`<d2l-menu-item-link 
+			text="${text}"
+			href="${href}"
+			target="${target}"
+			?preventDefault=${preventDefault}
+		></d2l-menu-item-link>`;
+	}
+	_createMenuItemMenu(node) {
+		const menuOpener =
+			node.querySelector('d2l-dropdown-button')
+			||  node.querySelector('d2l-dropdown-button-subtle');
+
+		const openerText = node.text || menuOpener.text;
+		const subMenu = node.querySelector('d2l-menu');
+		const subItems = [];
+
+		// iterate through any sub menus and turn them into menu items
+		for (const node of subMenu.children) {
+			subItems.push(this._convertToDropdownItem(node));
+		}
+		return html`<d2l-menu-item
+			text="${openerText}"
+		>
+			<d2l-menu>
+				${subItems}
+			</d2l-menu>
+		</d2l-menu-item>`;
+	}
+	_getLayoutItems(filteredNodes) {
+
+		const items = filteredNodes.map((node) => {
+			const computedStyles = window.getComputedStyle(node);
+			return {
+				type: node.tagName.toLowerCase(),
+				isChomped: false,
+				width: node.offsetWidth
+					+ parseInt(computedStyles.marginLeft.replace('px', ''))
```

Comment:
Nit: I've seen this just written as `parseInt(computedStyles.marginLeft) || 0` which works too.